### PR TITLE
Exceptions handled in Data Engineering

### PR DESCRIPTION
**Pull request by kkkarthik6**
Introduced exception handling.

### CODE REVIEW

File: src/nlpipe/pipelines/data_engineering/nodes.py [hunk 1]
```diff
@@ -5,47 +5,80 @@
...
         self.columns = columns
         self.target = target
-        # Shuffle all the data
+        if data.empty:
+            raise PipelineError('Please provide data', 'data argument empty, Cannot proceed without data')
+        if len(self.columns) < 1:
+            raise PipelineError('Please provide Input Column Name which has to be subjected to analysis', 'columns argument empty, Cannot proceed with out column name')
+        if len(self.target) < 1:
+            data['target']=[1]*data.shape[0]
+            warnings.warn("No target column provided, All data points will have same target value, Please use target argument to provide target labels", NoTargetWarning)
```
Comment (disssid) [inline src/nlpipe/pipelines/data_engineering/nodes.py:54]: Will a warning suffice or an error needs to be raised?

File: src/nlpipe/pipelines/data_engineering/nodes.py [hunk 2]
```diff
@@ -5,47 +5,80 @@
...
         # When returning many variables, it is a good practice to give them names:
         return dict(
-            train_x=self.train_x,
-            train_y=self.train_y,
-            test_x=self.test_x,
-            test_y=self.test_y,
+            train_x=train_x,
+            train_y=train_y,
+            test_x=test_x,
+            test_y=test_y,
```
Comment (disssid) [inline src/nlpipe/pipelines/data_engineering/nodes.py:79]: Unnecessary , at the end

File: src/nlpipe/pipelines/data_engineering/nodes.py [hunk 3]
```diff
@@ -5,47 +5,80 @@
 from typing import Any, Dict
 
 
+
```
Comment (disssid) [inline src/nlpipe/pipelines/data_engineering/nodes.py:8]: Add valid comments where ever possible to better understand the logic

**Review by disssid — CHANGES_REQUESTED** [review]

File: src/nlpipe/pipelines/data_engineering/nodes.py [hunk 4]
```diff
@@ -5,47 +5,80 @@
...
-            self.data = self.data.dropna(subset=target)
+        data = data
         self.columns = columns
         self.target = target
-        # Shuffle all the data
+        if data.empty:
+            raise PipelineError('Please provide data', 'data argument empty, Cannot proceed without data')
+        if len(self.columns) < 1:
+            raise PipelineError('Please provide Input Column Name which has to be subjected to analysis', 'columns argument empty, Cannot proceed with out column name')
+        if len(self.target) < 1:
```
[the remark this answers is not in the document]
Comment (kkkarthik6) [reply src/nlpipe/pipelines/data_engineering/nodes.py:52]: Done

File: src/nlpipe/pipelines/data_engineering/nodes.py [hunk 5]
```diff
@@ -5,47 +5,80 @@
...
-        self.data = data
-        if self.drop_nan:
-            self.data = self.data.dropna(subset=target)
+        data = data
         self.columns = columns
         self.target = target
-        # Shuffle all the data
+        if data.empty:
+            raise PipelineError('Please provide data', 'data argument empty, Cannot proceed without data')
+        if len(self.columns) < 1:
```
[the remark this answers is not in the document]
Comment (kkkarthik6) [reply src/nlpipe/pipelines/data_engineering/nodes.py:50]: Sure

File: src/nlpipe/pipelines/data_engineering/nodes.py (hunk 1 above)
Comment (kkkarthik6) [reply src/nlpipe/pipelines/data_engineering/nodes.py:54]: For unsupervised scenarios, we do not need labels, Hence I think warning would suffice.

File: src/nlpipe/pipelines/data_engineering/nodes.py (hunk 2 above)
Comment (kkkarthik6) [reply src/nlpipe/pipelines/data_engineering/nodes.py:79]: Trying to return dictionary, Hence key value pairs. It doesn't have to be a dictionary though, What would you suggest? return dictionary or tuple

Comment (disssid) [reply src/nlpipe/pipelines/data_engineering/nodes.py:79]: a dictionary sounds good

**Pull request merged**